### DISCUSSION TITLE
Block Hopper Minecarts from entering claims and being able to steal from floating claimed chests

### DISCUSSION THREAD

**Discussion by rcmaehl** [GriefPrevention/GriefPrevention#985] [Ideas]
**As** a player
**I want** hopper minecarts to not be pushed into my claims
**Because** it allows players to steal from floating chests by the cart being pushed into and out of the claim.

Optional questions:

**Is your feature request related to a problem? Please describe.**
Yes, however as someone who is not running GriefPrevention I can not provide you log outputs as required for a bug report. I reported this issue to the server I play on 32 days ago. Their own developers have not fixed this issue. As part of responsible disclosure, I am now reporting this to you.

**Describe alternatives you've considered**
Hopper minecarts could be banned from the server. Hopper minecarts could be broken if not placed on rails.

**Comment by Jikoo** [comment]
Your proposed solution isn't viable - listening to entity movement would be very resource-intensive. The case you're describing also isn't real, at least not in vanilla. Hopper minecarts only can pull from hoppers.
It is technically possible to steal from a hopper, but given the constraints (hopper must be exactly 1 block above perfectly level ground) I think it's unlikely - players almost always cover hoppers, though a lot of people do tend to favor unnaturally flat land.

A viable solution would be tagging carts with an owner, and if the owner is present and cannot access the cart cannot pull.
GP could deal with most edge cases by just transferring ownership on trust/claim changes. Regarding dispensed carts, while GP isn't a block logging suite I suppose tracking dispenser owner in a similar fashion would be viable and could potentially lead to other enhancements in projectile handling. Preexisting carts would essentially have administrative access due to no tagged owner, I don't see a way around that that wouldn't potentially heavily disrupt existing systems. It's still possible for minecart powerusers to run into troubles, but unlikely.

**Comment by RoboMWM** [comment]
This is 100% addon territory, given that it is very easy to deal with hoppers in any form similar to dealing with animals and other entities. Also, nearly anything with hoppers is a performance hit. GP ain't touching this.

**Comment by Jikoo** [comment]
I honestly think tagging owner might not be super horrendous as long as it's implemented properly, but I definitely agree. The actual situation in which this is possible is so unlikely and niche that I don't thing GP should consider adding a listener for the InventoryMoveItemEvent without a massive code restructure.

I do think it would be neat for GP to include a system to make it easy for addons to tag ownership on carts/whatever else though, that could actually simplify projectile ownership tracking logic too.

**Comment by RhythmicSys** [comment]
Hopper minecarts definitely do pull from chests, I've had a lot of issues with people stealing from chests with hopper minecarts. Hopper minecarts pull anything even from a full block above it. It's a major issue on larger multiplayer servers. Our server limits the placing of hopper minecarts to staff, players must request to have one placed by staff for this very reason

**Comment by Jikoo** [comment]
Interesting, that's new. I just tested and you're correct about chests. I couldn't replicate pulling a full block above. I know for sure it didn't work several versions ago, but that may have been a Spigot bug.

Given a typical "mailbox" design of a chest on a fence, I'd say this is actually pretty serious and could affect a pretty high percentage of any server's player base.

**Comment by RoboMWM** [comment]
If there's an efficient way to implement this I'd be ok with it - but as far as I'm aware, anything related to hoppers is always intensive, so much so that Paper will attempt to auto-disable the logic for handling InventorMoveItemEvent if no plugin is listening to it (this may have changed since they've implemented this but that's how bad hopper logic has been in Minecraft).

**Comment by rcmaehl** [comment]
👀 When did this get moved to ideas?

**Reply by RoboMWM** [reply]
All feature/addon suggestions are ideas, we did a mass migration of all suggestions to discussions when the private beta became open.
